crates/madara/client/settlement_client/src/eth/event.rs

```diff
@@ -0,0 +1,214 @@
+use crate::eth::StarknetCoreContract::LogMessageToL2;
+use crate::messaging::CommonMessagingEventData;
+use alloy::contract::EventPoller;
+use alloy::rpc::types::Log;
+use alloy::transports::http::{Client, Http};
+use anyhow::Error;
+use futures::Stream;
+use starknet_types_core::felt::Felt;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+type StreamItem = Result<(LogMessageToL2, Log), alloy::sol_types::Error>;
+type StreamType = Pin<Box<dyn Stream<Item = StreamItem> + Send + 'static>>;
```

Comment:
These names are too generic, consider renaming them to something more explicit.
```suggestion
type EthStreamItem = Result<(LogMessageToL2, Log), alloy::sol_types::Error>;
type EthStreamType = Pin<Box<dyn Stream<Item = StreamItem> + Send + 'static>>;
```